client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External dependencies
+ */
+import { has } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import handlers from './wpcom';
+
+export const middleware = store => next => action =>
+	has( handlers, action.type )
+		? handlers[ action.type ].forEach( handler => handler( store, action ) )
```

Comment:
I wondered about this. It's actually simpler since we have `castArray()` in **lodash**. The reason I went this direction (which was like a 60/40 decision) is that it maintains consistency in the code style in a way which shows that there can be multiple handlers. I was anticipating the question, "why is it a list in these cases and not in other?" or maybe more pointedly, "what if I want to have multiple handlers respond?" when seeing code that doesn't have the list.

and yes, I agree that a single handler will almost certainly be the most common case, although I like the idea that we could have some utility handlers in the API middleware to allow for things like this…

```js
export default {
	[ REQUEST_SITE ]: [ activateFetchingStatus, requestSite ]
}
```

_very_ willing to adjust here after discussion weighing the costs and benefits. I'm very conscientious here about setting an example for future contributions since this PR creates an entirely new subsystem for the app